pyccel/codegen/printing/ccode.py

```diff
@@ -260,6 +260,7 @@ class CCodePrinter(CodePrinter):
                       (NativeInteger(),8)     : 'int64_t',
                       (NativeInteger(),2)     : 'int16_t',
                       (NativeInteger(),1)     : 'int8_t',
+                      (NativeInteger(),-2)    : 'int',
```

Comment:
Why `-2`?

Reply:
Why not ? I needed a number for a key. Usually 0 is used for things that don't require a precision (e.g strings/classes). Positive numbers are used for actual precisions (as they would be printed in Fortran). -1 is used for the default Python precision. -2 seemed to be the next logical thing to use for the C default precision. It would probably be better with a name but I'm not sure how easy that would be to fit alongside what we currently have

If you have a better suggestion I'm so ears :)

Reply:
Why not indeed? 😄 But where do you need the C default integers exactly?

Reply:
Python seems to be quite picky about the signature of the `__init__` function (https://docs.python.org/3/extending/newtypes_tutorial.html#adding-data-and-methods-to-the-basic-example). It expects 
```c
int
Custom_init(CustomObject *self, PyObject *args, PyObject *kwds)
```

On most computers I think we could get away with using precision=4 because it's usually mapped to the same datatype under the hood but this can't be guaranteed

Reply:
There was a similar problem for `Exit` in Fortran : https://github.com/pyccel/pyccel/issues/1554

There we fixed it by handling this print as a special case. But this solution could also be used there.